File: Factorial.rb
class Integer
  def factorial
    if self==0
      1
    else
      return self*(self-1).factorial
    end
  end
end

puts "7!: #{7.factorial}"
puts "100!: #{100.factorial}"
puts "30!: #{30.factorial}"
puts "0!: #{0.factorial}"

File: Factorial.java
class Factorial{
  public static long factorial(int n){
    if (n==0) {
      return 1;
    }
    else{
      return n*factorial(n-1);
    }
  }
  public static void main(String[] args){
    System.out.println(factorial(6));
    System.out.println(factorial(0));
    System.out.println(factorial(3));
    System.out.println(factorial(30));
  }
  
}